octodns/equality.py

```diff
@@ -0,0 +1,30 @@
+#
+#
+#
+
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+
+class EqualityTupleMixin:
+
+    def _equality_tuple(self):
+        raise NotImplementedError('_equality_tuple method not implemented')
+
+    def __eq__(self, other):
+        return self._equality_tuple() == other._equality_tuple()
+
+    def __ne__(self, other):
+        return self._equality_tuple() != other._equality_tuple()
+
+    def __lt__(self, other):
+        return self._equality_tuple() < other._equality_tuple()
+
+    def __le__(self, other):
+        return self._equality_tuple() <= other._equality_tuple()
+
+    def __gt__(self, other):
+        return self._equality_tuple() > other._equality_tuple()
+
+    def __ge__(self, other):
+        return self._equality_tuple() >= other._equality_tuple()
```

Comment:
@theojulienne after looking at the equality stuff for a second I realized that I might as well pull this mixin out and DRY things up a bit. Looked around to see if there were any stdlib things like this and didn't find anything. It seems to work/make sense.

Reply:
functools.cmp_to_key and functools.total_ordering?